arcgis-ios-sdk-samples/Edit data/Edit and sync features/EditAndSyncFeaturesViewController.swift

```diff
@@ -52,19 +47,18 @@ class EditAndSyncFeaturesViewController: UIViewController {
     private var geodatabaseSyncTask: AGSGeodatabaseSyncTask!
     private var geodatabase: AGSGeodatabase!
     private var areaOfInterest: AGSEnvelope!
-
+    
     private var selectedFeature: AGSFeature? {
         didSet {
-            if let feature = selectedFeature {
-                if let featureLayer = feature.featureTable?.layer as? AGSFeatureLayer {
-                    featureLayer.select(feature)
-                }
+            if let feature = selectedFeature,
+               let featureLayer = feature.featureTable?.layer as? AGSFeatureLayer {
```

Comment:
This change means that if the selected feature's table's layer is not an AGSFeatureLayer, then the selection will not be cleared. Is that possible?

Reply:
My understanding might be wrong, but...

- Without making this change, if the `selectedFeature` is non-nil but its `featureTable.layer` is not an `AGSFeatureLayer`, nothing will happen
- With this change, if any of the 2 conditions is not met, then it would clear the selection.

It shouldn't make a difference in this sample as all the operational layers added are `AGSFeatureLayer`s. But I think it makes more sense to only allow 2 states:

1. Either a feature in an `AGSFeatureLayer` is selected
2. Or nothing selected

and avoid the state where `selectedFeature` is non-nil, but nothing happens in the UI.

Reply:
That makes sense. I think I just got confused.